#VRML_SIM R2021b utf8
WorldInfo {
  info [
    "Example use of a LightSensor device."
  ]
  title "Light Sensor"
  basicTimeStep 8
  coordinateSystem "NUE"
}
Viewpoint {
  orientation 0.9935553171106579 0.10488660901418098 -0.042972445714044585 5.436059563560858
  position -0.4269426384182182 3.004676353302797 2.726917082044323
}
TexturedBackground {
}
RectangleArena {
  floorSize 2 1.5
  floorAppearance Parquetry {
    colorOverride 0 1 1
  }
}
DEF LAMP Solid {
  translation 1 0.095 0
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0.8 0.8 0.8
        roughness 1
        metalness 0
        emissiveColor 1 1 1
      }
      geometry Sphere {
        radius 0.01
      }
    }
    PointLight {
      attenuation 0 0 5
      intensity 2
    }
  ]
  name "right"
}
DEF ROBOT Robot {
  translation 0.395525 0 -0.363572
  rotation 0 1 0 1.52166
  children [
    Transform {
      translation 0 0.0415 0
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.0820075 0.364731 0.8
            roughness 1
            metalness 0
          }
          geometry DEF BODY Cylinder {
            height 0.08
            radius 0.045
          }
        }
      ]
    }
    DistanceSensor {
      translation 0.02 0.03 -0.042
      rotation 0 1 0 1.07
      children [
        DEF distance_sensor Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds1_r"
      boundingObject USE distance_sensor
      physics Physics {
      }
    }
    DistanceSensor {
      translation -0.02 0.03 -0.042
      rotation 0 1 0 2.07
      children [
        DEF distance_sensor Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds0_l"
      boundingObject USE distance_sensor
      physics Physics {
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis -1 0 0
        anchor 0 0.025 0
      }
      device [
        RotationalMotor {
          name "left wheel motor"
          consumptionFactor 70
        }
        PositionSensor {
          name "left wheel sensor"
        }
      ]
      endPoint Solid {
        translation -0.045 0.025 0
        rotation 1 0 0 4.98467
        children [
          DEF WHEEL Transform {
            rotation 0 0 1 1.57
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 1 0 0
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.01
                  radius 0.025
                }
              }
            ]
          }
        ]
        name "left wheel"
        boundingObject USE WHEEL
        physics DEF PHYSICS_WHEEL Physics {
          density -1
          mass 0.05
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis -1 0 0
        anchor 0 0.025 0
      }
      device [
        RotationalMotor {
          name "right wheel motor"
          consumptionFactor 70
        }
        PositionSensor {
          name "right wheel sensor"
        }
      ]
      endPoint Solid {
        translation 0.045 0.025 0
        rotation 1 0 0 4.58735
        children [
          USE WHEEL
        ]
        name "right wheel"
        boundingObject USE WHEEL
        physics USE PHYSICS_WHEEL
      }
    }
    Transform {
      translation 0.02 0.063 -0.0385
      rotation 0 1 0 2.7489
      children [
        DEF UNUSED_EYE Group {
          children [
            Transform {
              translation 0 0 -1e-05
              rotation 0 0 1 0.8
              children [
                Shape {
                  appearance PBRAppearance {
                    baseColor 0.975691 0.981481 0.0252992
                    roughness 1
                    metalness 0
                  }
                  geometry Box {
                    size 0.02 0.005 0.005
                  }
                }
              ]
            }
            Transform {
              rotation 0 0 1 -0.8
              children [
                Shape {
                  appearance PBRAppearance {
                    baseColor 0.975691 0.981481 0.0252992
                    roughness 1
                    metalness 0
                  }
                  geometry Box {
                    size 0.02 0.005 0.005
                  }
                }
              ]
            }
          ]
        }
      ]
    }
    Transform {
      translation -0.02 0.063 -0.0385
      rotation 0 1 0 0.3927
      children [
        USE UNUSED_EYE
      ]
    }
    LightSensor {
      translation -0.02 0.07 -0.042
      rotation 0 1 0 2.07
      name "ls0"
      lookupTable [
        0 0 0
        4 1024 0
      ]
    }
    LightSensor {
      translation 0.02 0.07 -0.042
      rotation 0 1 0 1.07
      name "ls1"
      lookupTable [
        0 0 0
        4 1024 0
      ]
    }
    DEF SMILE Transform {
      translation 0 0.025 -0.036
      rotation 0.7745972408364944 0.44721228048294853 0.4472139204829502 1.8234794
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.721569 0.290196 0.290196
            roughness 1
            metalness 0
          }
          geometry Cylinder {
            height 0.018
            radius 0.009
            subdivision 3
          }
        }
      ]
    }
  ]
  name "MyBot"
  boundingObject Transform {
    translation 0 0.0415 0
    children [
      USE BODY
    ]
  }
  physics Physics {
    density -1
    mass 0.5
  }
  controller ""
}
DEF ROBOT Robot {
  translation -0.495525 0 -0.263572
  rotation 0 1 0 -1.52166
  children [
    Transform {
      translation 0 0.0415 0
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.0820075 0.364731 0.8
            roughness 1
            metalness 0
          }
          geometry DEF BODY Cylinder {
            height 0.08
            radius 0.045
          }
        }
      ]
    }
    DistanceSensor {
      translation 0.02 0.03 -0.042
      rotation 0 1 0 1.07
      children [
        DEF distance_sensor Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds1_r"
      boundingObject USE distance_sensor
      physics Physics {
      }
    }
    DistanceSensor {
      translation -0.02 0.03 -0.042
      rotation 0 1 0 2.07
      children [
        DEF distance_sensor Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds0_l"
      boundingObject USE distance_sensor
      physics Physics {
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis -1 0 0
        anchor 0 0.025 0
      }
      device [
        RotationalMotor {
          name "left wheel motor"
          consumptionFactor 70
        }
        PositionSensor {
          name "left wheel sensor"
        }
      ]
      endPoint Solid {
        translation -0.045 0.025 0
        rotation 1 0 0 4.98467
        children [
          DEF WHEEL Transform {
            rotation 0 0 1 1.57
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 1 0 0
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.01
                  radius 0.025
                }
              }
            ]
          }
        ]
        name "left wheel"
        boundingObject USE WHEEL
        physics DEF PHYSICS_WHEEL Physics {
          density -1
          mass 0.05
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis -1 0 0
        anchor 0 0.025 0
      }
      device [
        RotationalMotor {
          name "right wheel motor"
          consumptionFactor 70
        }
        PositionSensor {
          name "right wheel sensor"
        }
      ]
      endPoint Solid {
        translation 0.045 0.025 0
        rotation 1 0 0 4.58735
        children [
          USE WHEEL
        ]
        name "right wheel"
        boundingObject USE WHEEL
        physics USE PHYSICS_WHEEL
      }
    }
    Transform {
      translation 0.02 0.063 -0.0385
      rotation 0 1 0 2.7489
      children [
        DEF UNUSED_EYE Group {
          children [
            Transform {
              translation 0 0 -1e-05
              rotation 0 0 1 0.8
              children [
                Shape {
                  appearance PBRAppearance {
                    baseColor 0.975691 0.981481 0.0252992
                    roughness 1
                    metalness 0
                  }
                  geometry Box {
                    size 0.02 0.005 0.005
                  }
                }
              ]
            }
            Transform {
              rotation 0 0 1 -0.8
              children [
                Shape {
                  appearance PBRAppearance {
                    baseColor 0.975691 0.981481 0.0252992
                    roughness 1
                    metalness 0
                  }
                  geometry Box {
                    size 0.02 0.005 0.005
                  }
                }
              ]
            }
          ]
        }
      ]
    }
    Transform {
      translation -0.02 0.063 -0.0385
      rotation 0 1 0 0.3927
      children [
        USE UNUSED_EYE
      ]
    }
    LightSensor {
      translation -0.02 0.07 -0.042
      rotation 0 1 0 2.07
      name "ls0"
      lookupTable [
        0 0 0
        4 1024 0
      ]
    }
    LightSensor {
      translation 0.02 0.07 -0.042
      rotation 0 1 0 1.07
      name "ls1"
      lookupTable [
        0 0 0
        4 1024 0
      ]
    }
    DEF SMILE Transform {
      translation 0 0.025 -0.036
      rotation 0.7745972408364944 0.44721228048294853 0.4472139204829502 1.8234794
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.721569 0.290196 0.290196
            roughness 1
            metalness 0
          }
          geometry Cylinder {
            height 0.018
            radius 0.009
            subdivision 3
          }
        }
      ]
    }
  ]
  name "MyBot(1)"
  boundingObject Transform {
    translation 0 0.0415 0
    children [
      USE BODY
    ]
  }
  physics Physics {
    density -1
    mass 0.5
  }
  controller ""
}
DEF ROBOT Robot {
  translation 0.595525 0 0.163572
  rotation 0 1 0 1.52166
  children [
    Transform {
      translation 0 0.0415 0
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.0820075 0.364731 0.8
            roughness 1
            metalness 0
          }
          geometry DEF BODY Cylinder {
            height 0.08
            radius 0.045
          }
        }
      ]
    }
    DistanceSensor {
      translation 0.02 0.03 -0.042
      rotation 0 1 0 1.07
      children [
        DEF distance_sensor Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds1_r"
      boundingObject USE distance_sensor
      physics Physics {
      }
    }
    DistanceSensor {
      translation -0.02 0.03 -0.042
      rotation 0 1 0 2.07
      children [
        DEF distance_sensor Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds0_l"
      boundingObject USE distance_sensor
      physics Physics {
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis -1 0 0
        anchor 0 0.025 0
      }
      device [
        RotationalMotor {
          name "left wheel motor"
          consumptionFactor 70
        }
        PositionSensor {
          name "left wheel sensor"
        }
      ]
      endPoint Solid {
        translation -0.045 0.025 0
        rotation 1 0 0 4.9846699999999995
        children [
          DEF WHEEL Transform {
            rotation 0 0 1 1.57
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 1 0 0
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.01
                  radius 0.025
                }
              }
            ]
          }
        ]
        name "left wheel"
        boundingObject USE WHEEL
        physics DEF PHYSICS_WHEEL Physics {
          density -1
          mass 0.05
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis -1 0 0
        anchor 0 0.025 0
      }
      device [
        RotationalMotor {
          name "right wheel motor"
          consumptionFactor 70
        }
        PositionSensor {
          name "right wheel sensor"
        }
      ]
      endPoint Solid {
        translation 0.045 0.025 0
        rotation 1 0 0 4.58735
        children [
          USE WHEEL
        ]
        name "right wheel"
        boundingObject USE WHEEL
        physics USE PHYSICS_WHEEL
      }
    }
    Transform {
      translation 0.02 0.063 -0.0385
      rotation 0 1 0 2.7489
      children [
        DEF UNUSED_EYE Group {
          children [
            Transform {
              translation 0 0 -1e-05
              rotation 0 0 1 0.8
              children [
                Shape {
                  appearance PBRAppearance {
                    baseColor 0.975691 0.981481 0.0252992
                    roughness 1
                    metalness 0
                  }
                  geometry Box {
                    size 0.02 0.005 0.005
                  }
                }
              ]
            }
            Transform {
              rotation 0 0 1 -0.8
              children [
                Shape {
                  appearance PBRAppearance {
                    baseColor 0.975691 0.981481 0.0252992
                    roughness 1
                    metalness 0
                  }
                  geometry Box {
                    size 0.02 0.005 0.005
                  }
                }
              ]
            }
          ]
        }
      ]
    }
    Transform {
      translation -0.02 0.063 -0.0385
      rotation 0 1 0 0.3927
      children [
        USE UNUSED_EYE
      ]
    }
    LightSensor {
      translation -0.02 0.07 -0.042
      rotation 0 1 0 2.07
      name "ls0"
      lookupTable [
        0 0 0
        4 1024 0
      ]
    }
    LightSensor {
      translation 0.02 0.07 -0.042
      rotation 0 1 0 1.07
      name "ls1"
      lookupTable [
        0 0 0
        4 1024 0
      ]
    }
    DEF SMILE Transform {
      translation 0 0.025 -0.036
      rotation 0.7745972408364944 0.44721228048294853 0.4472139204829502 1.8234794
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.721569 0.290196 0.290196
            roughness 1
            metalness 0
          }
          geometry Cylinder {
            height 0.018
            radius 0.009
            subdivision 3
          }
        }
      ]
    }
  ]
  name "MyBot(2)"
  boundingObject Transform {
    translation 0 0.0415 0
    children [
      USE BODY
    ]
  }
  physics Physics {
    density -1
    mass 0.5
  }
  controller ""
}
DEF ROBOT Robot {
  translation -0.75 0 0.063572
  rotation 0 1 0 -0.6053603061004253
  children [
    Transform {
      translation 0 0.0415 0
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.0820075 0.364731 0.8
            roughness 1
            metalness 0
          }
          geometry DEF BODY Cylinder {
            height 0.08
            radius 0.045
          }
        }
      ]
    }
    DistanceSensor {
      translation 0.02 0.03 -0.042
      rotation 0 1 0 1.07
      children [
        DEF distance_sensor Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds1_r"
      boundingObject USE distance_sensor
      physics Physics {
      }
    }
    DistanceSensor {
      translation -0.02 0.03 -0.042
      rotation 0 1 0 2.07
      children [
        DEF distance_sensor Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds0_l"
      boundingObject USE distance_sensor
      physics Physics {
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis -1 0 0
        anchor 0 0.025 0
      }
      device [
        RotationalMotor {
          name "left wheel motor"
          consumptionFactor 70
        }
        PositionSensor {
          name "left wheel sensor"
        }
      ]
      endPoint Solid {
        translation -0.044999999999999984 0.025 0
        rotation 1 0 0 4.984670000000005
        children [
          DEF WHEEL Transform {
            rotation 0 0 1 1.57
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 1 0 0
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.01
                  radius 0.025
                }
              }
            ]
          }
        ]
        name "left wheel"
        boundingObject USE WHEEL
        physics DEF PHYSICS_WHEEL Physics {
          density -1
          mass 0.05
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis -1 0 0
        anchor 0 0.025 0
      }
      device [
        RotationalMotor {
          name "right wheel motor"
          consumptionFactor 70
        }
        PositionSensor {
          name "right wheel sensor"
        }
      ]
      endPoint Solid {
        translation 0.04499999999999997 0.025 0
        rotation 1 0 0 4.58735
        children [
          USE WHEEL
        ]
        name "right wheel"
        boundingObject USE WHEEL
        physics USE PHYSICS_WHEEL
      }
    }
    Transform {
      translation 0.02 0.063 -0.0385
      rotation 0 1 0 2.7489
      children [
        DEF UNUSED_EYE Group {
          children [
            Transform {
              translation 0 0 -1e-05
              rotation 0 0 1 0.8
              children [
                Shape {
                  appearance PBRAppearance {
                    baseColor 0.975691 0.981481 0.0252992
                    roughness 1
                    metalness 0
                  }
                  geometry Box {
                    size 0.02 0.005 0.005
                  }
                }
              ]
            }
            Transform {
              rotation 0 0 1 -0.8
              children [
                Shape {
                  appearance PBRAppearance {
                    baseColor 0.975691 0.981481 0.0252992
                    roughness 1
                    metalness 0
                  }
                  geometry Box {
                    size 0.02 0.005 0.005
                  }
                }
              ]
            }
          ]
        }
      ]
    }
    Transform {
      translation -0.02 0.063 -0.0385
      rotation 0 1 0 0.3927
      children [
        USE UNUSED_EYE
      ]
    }
    LightSensor {
      translation -0.02 0.07 -0.042
      rotation 0 1 0 2.07
      name "ls0"
      lookupTable [
        0 0 0
        4 1024 0
      ]
    }
    LightSensor {
      translation 0.02 0.07 -0.042
      rotation 0 1 0 1.07
      name "ls1"
      lookupTable [
        0 0 0
        4 1024 0
      ]
    }
    DEF SMILE Transform {
      translation 0 0.025 -0.036
      rotation 0.7745972408364944 0.44721228048294853 0.4472139204829502 1.8234794
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.721569 0.290196 0.290196
            roughness 1
            metalness 0
          }
          geometry Cylinder {
            height 0.018
            radius 0.009
            subdivision 3
          }
        }
      ]
    }
  ]
  name "MyBot(3)"
  boundingObject Transform {
    translation 0 0.0415 0
    children [
      USE BODY
    ]
  }
  physics Physics {
    density -1
    mass 0.5
  }
  controller "world1_driven_by_light_distance"
}
